core/src/validation.jl

```diff
@@ -441,6 +441,39 @@ function valid_demand(
     return !errors
 end
 
+function valid_outlet_crest_level!(graph::MetaGraph, outlet::Outlet, basin::Basin)::Bool
+    errors = false
+    for (index, (id, crest)) in enumerate(zip(outlet.node_id, outlet.min_crest_level))
+        id_in = inflow_id(graph, id)
+        if id_in.type == NodeType.Basin
+            #validation:
+            basin_bottom_level = basin_bottom(basin, id_in)
```

Comment:
For now I will use basin_bottom(basin, id_in)[2]. Later we can always change it back

Reply:
Since we have been using the basin bottom in many validation, can the basin bottom be part of the basin node?

Reply:
Since we'd be duplicating data, I prefer to do that if either it leads to a significant performance improvement, or simpler code. I'm not sure either would be the case here.